### PR TITLE
perf(stream): make Project concurrent to accelerate async exprs

### PR DESCRIPTION
**Pull request by TennyZhuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

As title

Please do some benchmark @wangrunji0408 

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by wangrunji0408 — APPROVED** [review]
👀

File: src/batch/src/executor/project.rs [hunk 1]
```diff
@@ -43,26 +46,38 @@ impl Executor for ProjectExecutor {
...
-            let mut arrays = Vec::with_capacity(self.expr.len());
-            for expr in &mut self.expr {
-                let column = Column::new(expr.eval(&data_chunk).await?);
-                arrays.push(column);
-            }
-            let (_, vis) = data_chunk.into_parts();
-            let ret = DataChunk::new(arrays, vis);
-            yield ret
-        }
+    // #[try_stream(boxed, ok = DataChunk, error = RwError)]
```
Comment (BugenZhao) [inline src/batch/src/executor/project.rs:54]:
```suggestion
```

File: src/batch/src/executor/project.rs [hunk 2]
```diff
@@ -43,26 +46,38 @@ impl Executor for ProjectExecutor {
...
+    fn do_execute(self) -> impl Stream<Item = Result<DataChunk>> + 'static {
+        let Self { expr, child, .. } = self;
+        let expr: Arc<[Box<dyn Expression>]> = expr.into();
+        child
+            .execute()
+            .map(move |data_chunk| {
+                let expr = expr.clone();
+                async move {
+                    let data_chunk = data_chunk?;
+                    // let data_chunk = data_chunk.compact();
```
Comment (BugenZhao) [inline src/batch/src/executor/project.rs:64]:
```suggestion
```

**Review by BugenZhao — APPROVED** [review]

File: src/batch/src/executor/project.rs
Comment (xxchan) [inline src/batch/src/executor/project.rs:1]: 🤣
image

Comment (TennyZhuang) [reply src/batch/src/executor/project.rs:1]: 🤣

**Pull request merged**